### DISCUSSION TITLE
Q: Is it possible to launch an extension from a project (node.js) script?

### DISCUSSION THREAD

**Discussion by mhkeller** [microsoft/vscode-discussions#256] [Extension Development QnA] [answered]
I have a Node.js script that runs some code to compile some HTML and SVG charts. I'd like to be display these charts in a pane in the Explorer area of the editor. I've seen some VS Code plugins that take a local JSON or CSV file as input but I haven't seen anything that you can call dynamically from a scripting language. The API would look something like this:

```js
// Read in some data
const data = parseCsv(readFileSync('my-data.csv', 'utf-8'));

// Create the markup using an existing node library that does this
const markup = createChart(data, 'line', { x: 'myX', y: 'myY' });

// This line would cause the chart to show up in the editor as its own window
showChart(markup);
```

Are there any examples showing how to do this or is it not currently possible? Thanks for the help.

**Comment by RandomFractals** [comment, marked as answer]
@mhkeller You can see how I do it in Vega viewer extension to load vega and vega-lite specs and display charts in Vega Preview webview:

https://github.com/RandomFractals/vscode-vega-viewer/blob/master/src/extension.ts#L76

You can also launch custom extensions and views via vscode Urls and Uri handler. See how it is implemented in Tabular data viewer in this ticket:

https://github.com/RandomFractals/tabular-data-viewer/issues/96

**Reply by mhkeller** [reply]
Thanks!
